superbench/benchmarks/micro_benchmarks/cuda_nccl_bw_performance.py

```diff
@@ -0,0 +1,181 @@
+        )
+        self._parser.add_argument(
+            '-c',
+            type=int,
+            default=0,
+            help='Check correctness of results. This can be quite slow on large numbers of GPUs. E.g. 0 or 1.',
+        )
+
+    def _preprocess(self):
+        """Preprocess/preparation operations before the benchmarking.
+
+        Return:
+            True if _preprocess() succeed.
+        """
+        if not super()._preprocess():
+            return False
+
+        # Format the arguments
+        if not isinstance(self._args.algo, list):
+            self._args.algo = [self._args.algo]
+        self._args.algo = [p.lower() for p in self._args.algo]
+
+        # Check the arguments and generate the commands
+        for algo in self._args.algo:
+            if algo not in self.__algorithms:
+                self._result.set_return_code(ReturnCode.INVALID_ARGUMENT)
+                logger.error(
+                    'Unsupported algorithm of NCCL test - benchmark: {}, algorithm: {}, expected: {}.'.format(
+                        self._name, algo, ' '.join(list(self.__algorithms.keys()))
+                    )
+                )
+                return False
+            else:
+                self._bin_name = self.__algorithms[algo]
+                if not self._set_binary_path():
+                    return False
+
+                command = os.path.join(self._args.bin_dir, self._bin_name)
+                command += ' -b {} -e {} -f {} -g {} -c {}'.format(
+                    self._args.min_size, self._args.max_size, str(self._args.f), str(self._args.gpu_count),
+                    str(self._args.c)
+                )
+                self._commands.append(command)
+
+        return True
+
+    def _process_raw_result(self, cmd_idx, raw_output):    # noqa: C901
+        """Function to parse raw results and save the summarized results.
+
+          self._result.add_raw_data() and self._result.add_result() need to be called to save the results.
+
+        Args:
+            cmd_idx (int): the index of command corresponding with the raw_output.
+            raw_output (str): raw output string of the micro-benchmark.
+
+        Return:
+            True if the raw output string is valid and result can be extracted.
+        """
+        self._result.add_raw_data('raw_output_' + self._args.algo[cmd_idx], raw_output)
+
+        content = raw_output.splitlines()
+        busbw_out = -1
+        time_out = -1
+        algbw_out = -1
+        try:
+            # Filter useless output
+            out_of_place_index = -1
+            out_of_bound_index = -1
+            for index, line in enumerate(content):
+                if 'out-of-place' in line:
+                    out_of_place_index = index
+                if 'Out of bounds values' in line:
+                    out_of_bound_index = index
+            content = content[out_of_place_index + 1:out_of_bound_index]
```

Comment:
Still the concern: If content is empty, we should return False, correct? I am not sure.

Reply:
Uh, this is prepared for MPI mode In fact for current implementation, when it has content for main process, it will have valid result, when it is empty for slave process, it will be empty. But If I know How Yifan implement MPI mode, I will add some check with the environment variable passed by him.